pkg/authorization/authorizer/authorizer.go

```diff
@@ -360,6 +377,48 @@ func (a DefaultAuthorizationAttributes) GetUserInfo() user.Info {
 func (a DefaultAuthorizationAttributes) GetVerb() string {
 	return a.Verb
 }
+
+// nonResourceMatches take the remainer of a URL and attempts to match it against a series of explicitly allowed steps that can end in a wildcard
+func (a DefaultAuthorizationAttributes) nonResourceMatches(rule authorizationapi.PolicyRule) bool {
+	for allowedNonResourcePath := range rule.NonResourceURLs {
+		allowPathSteps := splitPath(allowedNonResourcePath)
```

Comment:
Because my initial implementation allowed full regexes on each step and I failed to reconsider.  I'll tidy it up.